src/axom/inlet/Table.hpp

```diff
@@ -157,6 +157,24 @@ struct is_inlet_primitive_dict<std::unordered_map<VariantKey, T>>
   static constexpr bool value = is_inlet_primitive<T>::value;
 };
 
+template <typename T>
+struct is_std_vector : std::false_type
+{ };
+
+template <typename T>
+struct is_std_vector<std::vector<T>> : std::true_type
```

Comment:
In addition to the type parameter, std::vector has the allocator parameter. Given that Axom is going to be constructing the vector, its probably fine that it is restricted to the default allocator. It would probably be good to note that here, though. It might also be worth just adding the allocator type as a template parameter.

Reply:
I don't think we'd want to complicate the interface in that way.

I was actually going to suggest that, in the future, this move from `std::vector` to `axom::Array`, since we're trying to eliminate usage of `std::vector` from our codebase.  Note that axom::Array uses umpire allocations instead of std::allocators.

Tag: @bmhan12